### PR TITLE
[semver:minor] add workflow name parameter

### PR DESCRIPTION
**Pull request by mandos**
This is a quite naive approach to the problem in #7 

ps. I'm not JS dev so the code is ugly, sorry.

### CODE REVIEW

**Comment by mandos** [comment]
Anyone? @meeroslav ?

File: src/commands/set-shas.yml [hunk 1]
```diff
@@ -13,17 +13,25 @@ parameters:
     description: |
       By default, if no successful workflow is found on the main branch to determine the SHA,
       we will log a warning and use HEAD~1. Enable this option to error and exit instead.
+  workflow-name:
+    type: string
+    default: ""
+    description: |
+      By default, the script is looking for the last successful job. This param, make script to look
```
Comment (meeroslav) [inline src/commands/set-shas.yml:20]: Please rephrase this to (PR open on your repo for the convenience):
```
By default, the script is looking for the last successful job across all workflows. 
Set this param to search for the last successful job within a specific workflow.
```

File: src/examples/custom.yml [hunk 2]
```diff
@@ -2,6 +2,7 @@ description: >
   You need to specify `main-branch-name` if different than `main`.
   If last successful workflow run was not found, by default we report warning and fallback to HEAD~1.
   You can instead make this a hard error by settting 'error-on-no-successful-workflow' to true.
+  If you need to find a successful workflow with a specific name, set the value of 'workflow-name'.
```
Comment (meeroslav) [inline src/examples/custom.yml:5]: Please rephrase this to (PR open on your repo for convenience):
```
By default, the script is looking for the last successful job across all workflows. 
Set this param to search for the last successful job within a specific workflow.
```

File: src/scripts/find-successful-workflow.js [hunk 3]
```diff
@@ -92,9 +92,20 @@ function commitExists(commitSha) {
   }
 }
 
-async function isWorkflowSuccessful(pipelineId) {
-  return getJson(`https://circleci.com/api/v2/pipeline/${pipelineId}/workflow`)
-    .then(({ items }) => items.every(item => item.status === 'success'));
+async function isWorkflowSuccessful(pipelineId, workflowName) {
+  if (!workflowName || workflowName.length === 0) {
```
Comment (meeroslav) [inline src/scripts/find-successful-workflow.js:96]: `!workflowName` and `workflowName.length === 0` are equivalent for strings.
Use just the first condition.

File: src/scripts/find-successful-workflow.js [hunk 4]
```diff
@@ -92,9 +92,20 @@ function commitExists(commitSha) {
...
+    return getJson(`https://circleci.com/api/v2/pipeline/${pipelineId}/workflow`)
+      .then(({ items }) => items.every(item => item.status === 'success'));
+  } else {
+    return getJson(`https://circleci.com/api/v2/pipeline/${pipelineId}/workflow`).then(
+      function(items){
+        return items.items.some(
```
Comment (meeroslav) [inline src/scripts/find-successful-workflow.js:102]: Instead of `Array.some`, we should use `Array.find` first to find workflow based on name (each pipeline should have just a single workflow run) and then check for success. 

```JavaScript
return getJson(`https://circleci.com/api/v2/pipeline/${pipelineId}/workflow`).then(
  ({ items }) => {
     const workflowRun = items.find(item => item.name === workflowName);
     return workflowRun && workflowRun.status === 'success'
   });
```

**Review by meeroslav — CHANGES_REQUESTED** [review]
Thank you for the PR. Please see my comments.

**Comment by meeroslav** [comment]
That's fine. I applied them afterwards
